server/src/main/java/com/sauron/service/TransactionServiceImpl.java

```diff
@@ -52,17 +52,15 @@ public TransactionServiceImpl(RestTemplateBuilder restTemplateBuilder, UserRepos
         return transactions.stream().peek(t -> t.setBankId(bank.getId())).collect(Collectors.toList());
     }
 
-    private Collection<Transaction> fetchTransactions(Bank bankView, Long userId) {
-        Collection<Transaction> transactions = restTemplate.exchange(
-                createRequestEntity(bankView.getTransactionUrl(), userId),
-                RESPONSE_TYPE)
-                .getBody();
+    private Collection<Transaction> fetchTransactions(Bank bank, Long userId) {
+        Optional<String> apiUrl = bank.getBankApiUrl(GET_TRANSACTIONS);
+
+        Collection<Transaction> transactions = apiUrl.map(url -> restTemplate.exchange(
+                createRequestEntity(url, userId),
+                RESPONSE_TYPE).getBody())
+                .orElse(Collections.emptyList());
 
         return transactions != null ? transactions : Collections.emptyList();
```

Comment:
Actually, now it's never null (Optional guarantees it), so you are performing this logic twice :)

Reply:
Correct. It supposed to be inside mapper function which is now fixed.